--- a/Makefile
+++ b/Makefile
@@ -29,7 +29,7 @@ include Makeconfig
 
 # This is the default target; it makes everything except the tests.
 .PHONY: all
-all: lib others
+all: link lib others
 
 ifneq ($(AUTOCONF),no)
 
@@ -49,6 +49,9 @@ mv -f $@.new $@
 $(autoconf-it-cvs)
 endef
 
+link:
+	ln -vs libgcc.a `$(CC) -print-libgcc-file-name | sed 's/libgcc/&_eh/'`
+
 configure: configure.in aclocal.m4; $(autoconf-it)
 %/configure: %/configure.in aclocal.m4; $(autoconf-it)
 %/preconfigure: %/preconfigure.in aclocal.m4; $(autoconf-it)
