--- a/package/glibc/glibc.mk
+++ b/package/glibc/glibc.mk
@@ -59,6 +59,10 @@ ifeq ($(BR2_ENABLE_DEBUG),y)
 GLIBC_EXTRA_CFLAGS += -g
 endif
 
+ifeq ($(BR2_GCC_VERSION_8_X),y)
+GLIBC_EXTRA_CFLAGS += -Wno-error=attribute-alias
+endif
+
 # The stubs.h header is not installed by install-headers, but is
 # needed for the gcc build. An empty stubs.h will work, as explained
 # in http://gcc.gnu.org/ml/gcc/2002-01/msg00900.html. The same trick
