--- a/Config.in.legacy
+++ b/Config.in.legacy
@@ -233,15 +233,6 @@ config BR2_PACKAGE_KODI_VISUALISATION_FO
 	  According to upstream 'the visualization is not currently
 	  in a working shape.'
 
-config BR2_PACKAGE_PORTMAP
-	bool "portmap has been removed"
-	select BR2_LEGACY
-	select BR2_PACKAGE_RPCBIND
-	help
-	  The portmap upstream tarball is removed, no releases since
-	  ten years and latest change in upstream git in 2014.
-	  You should better use rpcbind as a RPC portmapper.
-
 config BR2_BINUTILS_VERSION_2_25_X
 	bool "binutils version 2.25 support removed"
 	select BR2_LEGACY
